Sources/Datadog/Tracing/DDSpan.swift

```diff
@@ -121,3 +121,23 @@ internal class DDSpan: OTSpan {
         )
     }
 }
+
+// MARK: - Logging errors
+
+internal extension OTSpan {
+    func logError(_ error: Error, timestamp: Date = Date()) {
+        var logFields = [
+            OTLogFields.event: "error",
+            OTLogFields.errorKind: "Unknown",
+            OTLogFields.message: "An error occurred",
+            OTLogFields.stack: String(describing: error)
+        ]
+        if let nsError = (error as NSError?) {
+            let errorKind = "\(nsError.domain) - \(nsError.code)"
+            logFields[OTLogFields.errorKind] = errorKind
+            let errorMessage = nsError.localizedDescription
+            logFields[OTLogFields.message] = errorMessage
+        }
+        log(fields: logFields, timestamp: timestamp)
+    }
```

Comment:
Marking `span` error this way will only work if Logging feature is enabled. If it's disabled, tracing auto-instrumentation will result with no error reporting.

Instead of depending on Logging For Tracing integration, we can set appropriate tags directly on `span`. This is exactly how LFT works:

https://github.com/DataDog/dd-sdk-ios/blob/c93f9687a7e18199d91f6f38e1a9944f1c3ed738/Sources/Datadog/Tracing/Span/SpanTagsReducer.swift#L54-L56

It's less code, and works independently of Logging feature.

Reply:
so we should prefer `DatadogTagKeys` over `OTLogFields`, is this correct?

Reply:
summary of a quick Zoom: yes, we prefer `DDTags` in internal code over `OTLogFields` ✅